File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.py
class Solution:
    def addSpaces(self, s: str, spaces: List[int]) -> str:
        answer=[]
        start=0
        for space in spaces:
            answer.append(s[start:space])
            start=space
        answer.append(s[start:])
        return " ".join(answer)

File: 2232-adding-spaces-to-a-string/adding-spaces-to-a-string.js
/**
 * @param {string} s
 * @param {number[]} spaces
 * @return {string}
 */
var addSpaces = function(s, spaces) {
    let answer=""
    let start=0;
    let space=0;
    while(start<s.length){
        if ((space<spaces.length) && (spaces[space]===start)){
            answer+=" ";
            space+=1;
            continue
            
        }
        else{
            answer+=s[start];
            start+=1;
        }
    }
    return answer;

    
};